src/board.py

```diff
@@ -81,3 +81,61 @@ def is_out_of_board(pawn):
 
 def is_out_of_base_line(coord):
     return coord < 0 or coord >= BASE_LINE_SIZE
+
+
+def is_crossable_right(pawn, fences):
+    top_right_fence = get_top_right_fence(pawn, fences)
+    bottom_right_fence = get_bottom_right_fence(pawn, fences)
+    return top_right_fence != Direction.VERTICALLY and bottom_right_fence != Direction.VERTICALLY
+
+
+def is_crossable_left(pawn, fences):
+    top_left_fence = get_top_left_fence(pawn, fences)
+    bottom_left_fence = get_bottom_left_fence(pawn, fences)
+    return top_left_fence != Direction.VERTICALLY and bottom_left_fence != Direction.VERTICALLY
+
+
+def is_crossable_up(pawn, fences):
+    top_left_fence = get_top_left_fence(pawn, fences)
+    top_right_fence = get_top_right_fence(pawn, fences)
+    return top_left_fence != Direction.HORIZONTALLY and top_right_fence != Direction.HORIZONTALLY
+
+
+def is_crossable_down(pawn, fences):
+    bottom_left_fence = get_bottom_left_fence(pawn, fences)
+    bottom_right_fence = get_bottom_right_fence(pawn, fences)
+    return bottom_left_fence != Direction.HORIZONTALLY and bottom_right_fence != Direction.HORIZONTALLY
+
+
+def get_top_left_fence(pawn, fences):
+    top_left_fence = Direction.NO
```

Comment:
If you check for the border case inside the is_crossable functions, you can directly return get_fence result.